apps/sms/views/inbox/js/inbox.js

```diff
@@ -622,18 +551,24 @@ var InboxView = {
           if (!Threads.has(draft.id)) {
             var thread = Thread.create(draft);
             Threads.set(draft.id, thread);
-            this.appendThread(thread);
+            promises.push(this.appendThread(thread));
           }
         }
       }
 
-      this.sticky && this.sticky.refresh();
+      return Promise.all(promises);
     });
   },
 
   prepareRendering: function inbox_prepareRendering() {
-    this.container.innerHTML = '';
-    this.renderDrafts();
+    // this.container.innerHTML = '';
+    this.listData = [];
+    this.list.configure({
+      getSectionName: function(item) {
+        return item.dayTextContent;
+      }
+    });
+    return this.prepareDrafts();
```

Comment:
I want to make sure the drafts data is ready before we start to render the thread, otherwise some edge case like draft is not ready while rendering data, and once the draft fetched it still could not be rendered because data is not inserted into list.